### DISCUSSION TITLE
gist demo to rename glyphs in .fea data, using feaLib

### DISCUSSION THREAD

**Discussion by justvanrossum** [fonttools/fonttools#3465] [Show and tell]
https://gist.github.com/justvanrossum/1f816294c55abd1e762976702b01e6b7

I could submit this as a Snippet, but it doesn't really have a useful main program. Would it perhaps still be useful as a snippet?

**Comment by anthrotype** [comment]
nice! I think we could even add this as a method on the feaLib.ast classes.
If you prefer to dump this in the Snippet directory, i'm ok with that too.

One of the reasons ufo2ft does the glyph production names renaming at the end, in its postProcessor, directly manipulating the binary TTFont's post or CFF tables, is because we didn't have a handy way to rename the glyph names in the features.fea.

**Reply by anthrotype** [reply]
a more useful `main()` could take a .csv file each line with `old,new` names and parse that into a renaming map and the renameFunc looks them up in there

**Reply by justvanrossum** [reply]
Or should we consider a new feaLib submodule for ast operations? The merge and subset logic from ufomerge would be logical candidates, too. I don't think we should bother ast.py with this.

**Reply by anthrotype** [reply]
makes sense, yes

**Reply by typoman** [reply]
FWIW, I wrote a subsetter that you can see in [fontgadgets](https://github.com/typoman/fontgadgets) features module. I really want to add it to fonttools, but requires writing a lot of tests. It is based on @anthrotype suggestion to add subset method to `ast` objects. I also needed extra data during subset, so I add them to feature objects using another parser subclass. Not sure if it's too complicated and can be done simpler, like what @justvanrossum did with rename the snippet.

Edit:
The last version of fontgadgets is inside the [RoboFontGadgets](https://github.com/typoman/RoboFontGadgets) extension.

**Reply by typoman** [reply]
Here is the direct link to the subsetter module:
https://github.com/typoman/RoboFontGadgets/blob/main/fontgadgets.roboFontExt/lib/fontGadgets/extensions/features/subset.py

**Reply by justvanrossum** [reply]
Have you compared your subsetter with the one in ufomerge?

**Reply by typoman** [reply]
I haven't, but from the code, ufomerge seems very elaborate. I will try it to see how it works on the fonts I do the production on.
